From cd90ea7130bb52bb579bb9c57e13976dcd23d003 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Sat, 13 Oct 2018 12:02:42 +0530
Subject: [PATCH 0470/1852] drm: xlnx: pl_disp: Enable early callback

commit 5b8d1ea20488f6ecdec8aa1d18e3898fae36b2a0 from
https://github.com/Xilinx/linux-xlnx.git

Patch adds support for enabling early callback. This enables the
user space applications like kmssink plugin that queue up only one
buffer at a time to not drop any frames due to timeout receiving
vblank event from the driver.
The side effect of enabling this behavior is that the buffer whose
callback function has been called may actually still be scanning
out.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_pl_disp.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_pl_disp.c b/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
index 28f5779ac6a2..55f7522873d2 100644
--- a/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
+++ b/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
@@ -179,6 +179,7 @@ static void xlnx_pl_disp_plane_enable(struct drm_plane *plane)
 	}
 	desc->callback = xlnx_pl_disp->callback;
 	desc->callback_param = xlnx_pl_disp->callback_param;
+	xilinx_xdma_set_earlycb(xlnx_dma_chan->dma_chan, desc, true);
 
 	if (plane->state->fb->flags == DRM_MODE_FB_ALTERNATE_TOP ||
 	    plane->state->fb->flags == DRM_MODE_FB_ALTERNATE_BOTTOM) {
-- 
2.31.1

